lib/rack/lint.rb

```diff
@@ -328,10 +323,20 @@ def check_environment(env)
           raise LintError, "rack.url_scheme unknown: #{env[RACK_URL_SCHEME].inspect}"
         end
 
-        ## * There must be a valid input stream in <tt>rack.input</tt>.
-        check_input env[RACK_INPUT]
+        ## * There may be a valid input stream in <tt>rack.input</tt>.
+        if rack_input = env[RACK_INPUT]
+          check_input_stream(rack_input)
+          @env[RACK_INPUT] = InputWrapper.new(rack_input)
+        end
+
         ## * There must be a valid error stream in <tt>rack.errors</tt>.
-        check_error env[RACK_ERRORS]
+        if rack_errors = env[RACK_ERRORS]
```

Comment:
Because of the `env missing required key check` above, this `if` is unnecessary, as `rack.errors` should always be set at this point.

Reply:
Thanks, fixed.